From 6b3e6a62228baba28332c1800571a804ad6cb2e3 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@kde.org>
Date: Wed, 31 Jul 2024 01:03:20 +0200
Subject: [PATCH] QQuickItem: Fix crash in mapToItem

ffb557ff17d45684aac159762ae4bcc8a04ba573 fixed crashing
when item->window() is null, but window() may also be null.

Pick-to: 6.8
Change-Id: I606dfc1e32cc6c095f92e5b1d2eae62b5a624cb7
---
 src/quick/items/qquickitem.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/quick/items/qquickitem.cpp b/src/quick/items/qquickitem.cpp
index e3d70851398..55766addddf 100644
--- a/src/quick/items/qquickitem.cpp
+++ b/src/quick/items/qquickitem.cpp
@@ -8774,8 +8774,9 @@ QPointF QQuickItem::mapToItem(const QQuickItem *item, const QPointF &point) cons
     QPointF p = mapToScene(point);
     if (item) {
         const QQuickWindow *itemWindow = item->window();
-        if (itemWindow != nullptr && itemWindow != window())
-            p = itemWindow->mapFromGlobal(window()->mapToGlobal(p));
+        const QQuickWindow *thisWindow = window();
+        if (thisWindow && itemWindow && itemWindow != thisWindow)
+            p = itemWindow->mapFromGlobal(thisWindow->mapToGlobal(p));
 
         p = item->mapFromScene(p);
     }
-- 
2.16.3
